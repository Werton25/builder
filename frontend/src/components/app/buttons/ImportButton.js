import React from 'react';
import PropTypes from 'prop-types';
import {importRequests} from "../../../actions/requests";
import {readJsonFile} from "../../../utils";
import {toastr} from "react-redux-toastr";
import {connect} from "react-redux";

export class BaseImportButton extends React.Component {
  onImportRequests = (e) => {
    const file = e.target.files[0];
    readJsonFile(file)
      .then(this.props.importRequests)
      .catch(err => toastr.error('Ошибка', err.message));
  };

  render() {
    return <input className='hidden' type='file' onChange={this.onImportRequests}/>
  }
}

BaseImportButton.propTypes = {
  importRequests: PropTypes.func.isRequired
};

const mapDispatchToProps = {
  importRequests
};

export default connect(null, mapDispatchToProps)(BaseImportButton);
