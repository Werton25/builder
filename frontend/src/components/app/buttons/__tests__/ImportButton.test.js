import React from "react";
import renderer from "react-test-renderer";
import ImportButton from "../ImportButton";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

describe('ImportButton tests', () => {
  it('ImportButton renders correctly', () => {
    const store = configureStore()({});
    const tree = renderer
      .create(
        <Provider store={store}>
          <ImportButton/>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
