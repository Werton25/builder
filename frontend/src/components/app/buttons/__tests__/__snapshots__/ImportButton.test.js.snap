// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportButton tests ImportButton renders correctly 1`] = `
<input
  className="hidden"
  onChange={[Function]}
  type="file"
/>
`;
